crates/weaver_resolved_schema/src/any_value.rs

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#![allow(rustdoc::invalid_html_tags)]
+
+//! Specification of a resolved `AnyValue`.
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use weaver_semconv::attribute::{EnumEntriesSpec, Examples, RequirementLevel};
+use weaver_semconv::stability::Stability;
+
+/// An `AnyValue` definition.
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq, Hash, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub struct AnyValue {
```

Comment:
If any values remain unchanged during the resolution process, why do we need to redefine `AnyValueSpec` in the `semconv` crate? I recommend removing this struct and using `AnyValueSpec` in the resolved registry instead. If, at some point, we have elements in any value that become resolvable, we can reintroduce this struct without breaking the contract with the jq and Jinja2 engines, as they only interact with the JSON/YAML external representation of the resolved registry.

Reply:
@lquerel, @jsuereth
Ok, I started doing this locally and I've hit the following
- `type` is mostly unaffected, the only difference is for the `enum` type as the resolved version (to be the same as the existing enum) includes the id's of the enums (i.e. `enum {active, inactive, background, foreground, terminate}`) where now it becomes just `enum` (probably not a bad thing, and maybe be the preferred approach anyway)
- Bigger (potential) issue, because we use `id` in the semconv and `name` in the resolved, by removing the resolved `AnyValue` it also becomes `id` -- Question is this what we want or do we want the resolved `AnyValue` to contain `name` instead of `id`?

Reply:
Advantages
- Same as `enum` member(s) are they are already just `id` and not `name`
- if we want to introduce a `name` to describe the `AnyValueSpec` later to describe the "type" of a `map` etc we could.
- Has more of a separation from the "groups" vs "values" (groups have both "id" and "name" where the name is really just the "id". So the "body" (or any location of an `AnyValue` will just have its original `id`

Reply:
Ok, I've updated this PR with the changes to "remove" the resolved and resolver `any_value.rs` files and usage, I'll wait for the answer to the above before publishing the Event PR with the change (I already have it locally and it's just waiting). I'll add some comments and link here to the "id" / "name" change for the test files for some context.

Reply:
Since we can still introduce names in the body fields later, I’ll go with the simpler version to minimize code duplication.